### PR TITLE
Support admin port in openapi

### LINKED ISSUE #893: MP OpenAPI should support "admin" port
The `OpenAPIMpService` should be configurable to run the routing on a different port than the default.

See approach taken in `HealthMpService` and `MetricsMpService`.

### PR DESCRIPTION
**Pull request by tjquinno**
Fixes #893 

This PR refactors the nearly-duplicate logic in the metrics and health MP services into a new interface and references it from those services and from OpenAPI.

### CODE REVIEW

File: microprofile/metrics/src/main/java/io/helidon/microprofile/metrics/MetricsMpService.java [hunk 1]
```diff
@@ -46,22 +45,12 @@ public void configure(MpServiceContext serviceContext) {
...
-        if (routingNameConfig.isPresent()) {
-            String routingName = routingNameConfig.get();
-            // support for overriding this back to default routing using config
-            if (!"@default".equals(routingName)) {
-                endpointRouting = serviceContext.serverNamedRoutingBuilder(routingName);
-            }
-        }
-
-        metricsSupport.configureVendorMetrics(null, defaultRouting);
+        metricsSupport.configureVendorMetrics(null, routingBuilders.defaultBuilder());
```
Comment (tomas-langer) [inline microprofile/metrics/src/main/java/io/helidon/microprofile/metrics/MetricsMpService.java]: The `null` as the first parameter is terrible, though I did that :(

File: microprofile/server/src/main/java/io/helidon/microprofile/server/RoutingBuilders.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                        .map(context::serverNamedRoutingBuilder)
+                        // use default server routing
+                        .orElse(defaultRoutingBuilder);
+        return new RoutingBuildersImpl(defaultRoutingBuilder, actualRoutingBuilder);
+    }
+
+    /**
+     * Package-private implementation of the {@code RoutingBuilders} interface.
+     */
+    class RoutingBuildersImpl implements RoutingBuilders {
```
Comment (tomas-langer) [inline microprofile/server/src/main/java/io/helidon/microprofile/server/RoutingBuilders.java]: As this class is defined within the interface, it is implicitly public. You need to move it to the package to make it package private.

File: microprofile/server/src/main/java/io/helidon/microprofile/server/RoutingBuilders.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    /**
+     * Prepares the default and actual {@link Routing.Builder} instances based
+     * on the "routing" configuration for the specific component.
+     *
+     * @param context the {@code MpServiceContext} for the calling service
+     * @param componentName config key under which "routing" config might exist for the component of interest
+     * @return {@code RoutingBuilders} based on the named config (or default)
+     */
+    static RoutingBuilders newRoutingBuilders(MpServiceContext context, String componentName) {
```
[the remark this answers is not in the document]
Comment (tjquinno) [reply microprofile/server/src/main/java/io/helidon/microprofile/server/RoutingBuilders.java]: fixed.

File: microprofile/server/src/main/java/io/helidon/microprofile/server/RoutingBuilders.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+ * Provides {@link Routing.Builder} instances (for the default and the actual)
+ * for a Helidon MP service, based on configuration for the component (if any)
+ * and defaults otherwise.
+ */
+public interface RoutingBuilders {
+
+    /**
+     *
+     * @return the default {@code Routing.Builder} for the component
+     */
```
[the remark this answers is not in the document]
Comment (tjquinno) [reply microprofile/server/src/main/java/io/helidon/microprofile/server/RoutingBuilders.java]: Fixed.

File: microprofile/server/src/main/java/io/helidon/microprofile/server/RoutingBuilders.java (hunk 2 above)
Comment (tjquinno) [reply microprofile/server/src/main/java/io/helidon/microprofile/server/RoutingBuilders.java]: Refactored.

File: microprofile/metrics/src/main/java/io/helidon/microprofile/metrics/MetricsMpService.java (hunk 1 above)
Comment (tjquinno) [reply microprofile/metrics/src/main/java/io/helidon/microprofile/metrics/MetricsMpService.java]: I stand on the shoulders of giants...

File: microprofile/metrics/src/main/java/io/helidon/microprofile/metrics/MetricsMpService.java [hunk 5]
```diff
@@ -46,22 +45,12 @@ public void configure(MpServiceContext serviceContext) {
 
         MetricsSupport metricsSupport = MetricsSupport.create(metricsConfig);
 
-        ConfigValue<String> routingNameConfig = metricsConfig.get("routing").asString();
-        Routing.Builder defaultRouting = serviceContext.serverRoutingBuilder();
+        RoutingBuilders routingBuilders = RoutingBuilders.create(
+                serviceContext, metricsConfig);
```
Comment (tomas-langer) [inline microprofile/metrics/src/main/java/io/helidon/microprofile/metrics/MetricsMpService.java]: Strange formatting - this should be on the same line

**Pull request merged**